R/augsynth.R

```diff
@@ -1,323 +1,74 @@
-
-    # ## post treatment estimate
-    # att_post <- colMeans(augsynth$data$y[augsynth$data$trt == 1,,drop=F]) -
-    #     predict(augsynth)
-
-    # ## pre treatment estimate
-    # att_pre <- colMeans(augsynth$data$X[augsynth$data$trt == 1,,drop=F]) -
-    #     t(augsynth$data$X[augsynth$data$trt==0,,drop=F]) %*% augsynth$weights
-    att_est <- predict(augsynth, att = T)
-    att <- data.frame(Time = augsynth$data$time,
-                        Estimate = att_est)
-
-
-    if(augsynth$progfunc == "Ridge" |
-       augsynth$progfunc == "None" & augsynth$scm) {
-        ridge <- augsynth$progfunc == "Ridge"
-        scm <- augsynth$scm
-
-        ## get standard errors
-        synth_data <- format_synth(augsynth$data$X, augsynth$data$trt,
-                                   augsynth$data$y)
-
-        if(jackknife) {
-            att_se <- jackknife_se_ridgeaug(augsynth$data, synth_data,
-                                            augsynth$data$Z, augsynth$lambda,
-                                            ridge, scm, augsynth$fixedeff)
-        } else {
-            att_se <- loo_se_ridgeaug(augsynth$data, synth_data, 
-                                      augsynth$data$Z,
-                                      augsynth$lambda,
-                                      ridge, scm)
-        }
-
-        att$Std.Error <- att_se$se
-
-        summ$att <- att
-        summ$sigma <- att_se$sigma
-    } else {
-        ## no standard errors
-        att$Std.Error <- NA
-        summ$att <- att
-        summ$sigma <- NA
+#' 
+augsynth <- function(form, unit, time, data, t_int=NULL, ...) {
+
+  call_name <- match.call()
+  
+  form <- Formula::Formula(form)
+  unit_quosure <- enquo(unit)
+  time_quosure <- enquo(time)
+  
+  ## format data
+  outcome <- terms(formula(form, rhs=1))[[2]]
+  trt <- terms(formula(form, rhs=1))[[3]]
+  wide <- format_data_stag(outcome, trt, unit_quosure, time_quosure, data=data)
+
+  trt_year = -1
+  multi = F
+  for (u in wide$trt) {
+    if (is.finite(u) && u != trt_year) {
+      if (trt_year == -1) {
+        trt_year = u
+      } else {
+        multi = T
+        break
+      }
     }
+  }
 
-    summ$t_int <- augsynth$t_int
-    summ$call <- augsynth$call
-    summ$l2_imbalance <- augsynth$l2_imbalance
-    summ$scaled_l2_imbalance <- augsynth$scaled_l2_imbalance
-    ## get estimated bias
-
-    if(augsynth$progfunc == "Ridge") {
-        mhat <- augsynth$ridge_mhat
-        w <- augsynth$synw
-    } else {
-        mhat <- augsynth$mhat
-        w <- augsynth$weights
-    }
-    trt <- augsynth$data$trt
-    m1 <- colMeans(mhat[trt==1,,drop=F])
-
-    summ$bias_est <- m1 - t(mhat[trt==0,,drop=F]) %*% w
-    if(augsynth$progfunc == "None" | (!augsynth$scm)) {
-        summ$bias_est <- NA
+  if (multi) {
+    return(multisynth(form, !!enquo(unit), !!enquo(time), data, ...)) 
+  } else {
+    if (is.null(t_int)) {
+      t_int = data %>% pull(!!time_quosure) %>% unique() %>% sort() %>% `[`(trt_year+1)
```

Comment:
Probably this works, but maybe just double check some edge cases, like if there's uneven spacing between time periods or something.

Reply:
In the single synth case this shouldn't be a problem because `trt_year` is an index of the year  and not the year itself.

Reply:
There may be potential other edge cases though, let me know if you think of anything; I think it seems good to me because the treatment year is consistently the same right?

Reply:
Yeah looks good to me